databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -394,16 +394,25 @@ def delete_share_cli(api_client, name):
               help=(
                   'IP address in CIDR notation that is allowed to use delta sharing. '
                   '(can be specified multiple times).'))
+@click.option('--custom-property', default=None, required=False, multiple=True,
+              help=(
+                  'Custom properties of the recipient. Key and value should be provided '
+                  'at the same time separated by an equal sign. '
+                  'Example: --custom-property country=US'))                 
 @debug_option
 @profile_option
 @eat_exceptions
 @provide_api_client
-def create_recipient_cli(api_client, name, comment, sharing_id, allowed_ip_address):
+def create_recipient_cli(api_client, name, comment, sharing_id,
+                         allowed_ip_address, custom_property):
     """
     Create a new recipient.
     """
+    parsed_custom_properties = [property_str.split('=', 2) for property_str in custom_property]
+    custom_properties = [{"key": property_key, "value": property_value}
```

Comment:
Should add check that the split actually created 2 entries and respond with appropriate error message if not (eg someone passing in a parameter that does not contain =)

Reply:
Added extra check.